File: backends/fiber/internal/models/step.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Step struct {
	ID              uuid.UUID        `gorm:"column:step_id;primaryKey;default:gen_random_uuid()" json:"id"`
	Title           string           `gorm:"column:title;not null" json:"title"`
	Instructions    string           `gorm:"column:instructions;not null" json:"instructions"`
	StepNumber      int32            `gorm:"column:step_number;not null" json:"step_number"`
	RecipeID        uuid.UUID        `gorm:"column:recipe_id" json:"-"`
	StepIngredients []StepIngredient `gorm:"foreignKey:StepID" json:"step_ingredients,omitempty"`
	CreatedAt       time.Time        `gorm:"column:created_at;default:CURRENT_TIMESTAMP" json:"created_at"`
	UpdatedAt       time.Time        `gorm:"column:updated_at;default:CURRENT_TIMESTAMP" json:"updated_at"`
}

File: backends/aspnet/Recipes.API/entities/Step.cs
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Recipes.API.entities;

public partial class Step
{
    [JsonPropertyName("id")]
    public Guid StepId { get; set; }

    public string Title { get; set; } = null!;

    public string Instructions { get; set; } = null!;

    public int StepNumber { get; set; }

    public Guid? RecipeId { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdatedAt { get; set; }

    public virtual Recipe? Recipe { get; set; }

    public virtual ICollection<StepIngredient> StepIngredients { get; set; } = new List<StepIngredient>();
}
